### DISCUSSION TITLE
Best way to push timeline depending on condition / loop through global variable "condition"

### DISCUSSION THREAD

**Discussion by vchopurian** [jspsych/jsPsych#1557] [Q&A]
Hi! First of all, thank you all so much for the fantastic work, you've made building an online experiment so much easier! My question might have already been asked, my apologies if this is the case. 

I have an experiment with auditory, visual and mixed trials, which show one auditory and one visual stimulus. Those trials should be counterbalanced based on a predefined arrays (getCondList). Further, I have two conditions, whether stimulus 1 or 2 has to be recalled. Depending on the modality of the selected stimulus, I wrote a separate plugin for each visual and auditory reconstruction (recall). This makes it 8 conditions in total (4 (trial type: auditory, visual, auditory-visual and visual-auditory) x 2 (recall stimulus: first or second)). Right now, I've build separate timelines for each of the conditions. Then, I wanted to build a loop for the total number of trials and put the timeline.push(conditionX) into the if statements. However, this seems to slow down the experiment a great deal. 

Looks something like this:
```
function trialLoop(nTrial,getCondList){
var whichCond = getCondList[i];
timeline.push(startFullScreen);

    if (whichCond == 1) {
        
        timeline.push(visualTrial);

    } else if(whichCond = 2) {

        timeline.push(auditoryTrial);
[...]
```

Is there any way I could build this a bit smarter? Could anyone give me a hint? Would be eternally grateful!

**Comment by jodeleeuw** [comment]
Hi @vchopurian,

When you say that it slows down the experiment, can you describe more what happens? This sounds like a pretty reasonable strategy to me, so I'm wondering if there's something in the details that is causing a problem.

**Reply by vchopurian** [reply]
Thank you so much for your reply, I think I fixed it and indeed, it was a problem with how I specified the loop. It works now! (It always just showed the blank grey screen, without showing an error in the console, so I just assumed it slows down the loading or something but oh well)
